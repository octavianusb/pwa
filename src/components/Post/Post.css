.title {
  font-size: 3rem;
}

.title a {
  color: #2f89de;
}
.title a:hover, .title a:focus {
  color: #30699e;
  transition: color 0.4s ease;
}

.content {
  font-size: 1.8rem;
  line-height: 1.7;
}
