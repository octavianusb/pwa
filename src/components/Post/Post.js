import React from 'react'
import { Link } from 'react-router-dom'

import styles from './Post.css'

export default function Post (props) {
  return (
    <div>
      <h2 className={styles.title}>
        {
          props.titleLink
          ? <Link to={`/post/${props.slug}`}>{props.title}</Link> : props.title
        }
      </h2>
      <p className={styles.content}>{props.excerpt}</p>
    </div>
  )
}
