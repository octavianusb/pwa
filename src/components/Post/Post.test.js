import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Post from './Post'

describe('Post component', () => {
  it('should render as expected', () => {
    const component = shallow(<Post title="My post" />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('should render with title type link', () => {
    const component = shallow(<Post title="My post" titleLink />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
