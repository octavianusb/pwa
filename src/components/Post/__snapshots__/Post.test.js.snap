// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should render as expected 1`] = `
<div>
  <h2
    className="title"
  >
    My post
  </h2>
  <p
    className="content"
  />
</div>
`;

exports[`Post component should render with title type link 1`] = `
<div>
  <h2
    className="title"
  >
    <Link
      replace={false}
      to="/post/undefined"
    >
      My post
    </Link>
  </h2>
  <p
    className="content"
  />
</div>
`;
